### ISSUE TITLE
Bug: passing dynamic parameters does not work when using and([]) in guard

### ISSUE THREAD

**Issue by melishev** [statelyai/xstate#5145]
### XState version

XState version 5

### Description

Hi, awesome library, thank you so much for your work!

I want to combine defenders in my event, for example like this:
```js
example: {
      on: {
        EVENT_NAME: {
          guard: and([
            'someGuard',
            {
              type: 'someGuard2',
              params: ({ event }) => ({ some: event.some }),
            },
          ]),
        },
      },
```
I decided to try this because I saw such a case study in your [test files](https://github.com/statelyai/xstate/blob/main/packages/core/test/guards.test.ts#L1352), but didn't check their execution

But in a case like this I run into two problems:
1. Typing doesn't work (suppose you don't accept such a case, then it's logical)
2. Guard is not called at all

If this case is not admissible, could you then highlight how I can solve my problem?

### Expected result

I expect the syntax I'm using will work with the new fix version

### Actual result

not work

### Reproduction

--

**Comment by melishev** [comment]
Sorry, it's my mistake, I tried your tests, tried my code and everything is ok.

The only problem is with types, this syntax breaks typing, are there any plans for this?

**Comment by roblabat** [comment]
Hello, is there any updates or someone working on this ? If not I can try to make a fix for this.

**Comment by davidkpiano** [comment]
cc. @Andarist 

Thanks for offering @roblabat - if you have an idea of how to fix this, we'd appreciate the help, although I fear it's probably a tricky typing issue.

**Comment by roblabat** [comment]
@davidkpiano @Andarist, I don't think this is purely a problem of typings here. Diving into the code of guards.ts I found quite strange the implementation of BuiltinGuard. They are implemented as functions that are never called but with an attribut. check that will be called in replacement to that function. This works because js is a compliant langage but I'm not sure of the pattern. My Idea of a fix would be to refactor builtin guards to return an InlineGuard function and clean the function evaluateGuard to remove those builtinGuards evaluation. Then making cleanning in typings of guards.

Before I start developments on this could tell me if this way of working sounds good for you or if there were a technical reason I'm missing for this builtinGuards implementation with the .check attribut ?

**Comment by davidkpiano** [comment]
I would defer to @Andarist for the technical reason for this.

**Comment by GeorgeGkas** [comment]
Hello everyone, I ran into this issue and wanted to share my findings. It looks like using the and guard breaks the return type of createMachine. For example, these guards cause problems:

```ts
and(["isCurrentPlayersTurn", "moveIsValid", "isFirstGameMove"])
```

All of the above are defined as parameterised functions like `({ context, event }) => {}`.

If you run into this, the only current workaround is to cast the guard call to `any` like so:

```ts
and(["isCurrentPlayersTurn", "moveIsValid", "isFirstGameMove"]) as any
```

 This isn’t ideal, but it restores the actor typings in most cases (though I’m not sure what happens when accessing deeply nested properties related to guards in the actor’s return value).

I also tried narrowing the type with `GuardPredicate`, but it seems we’re missing an exported `NormalizeGuardArgArray` type. Having that would let us provide all the necessary type parameters in `GuardPredicate`.
